apiml/src/main/java/org/zowe/apiml/filter/LogoutHandler.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * This program and the accompanying materials are made available under the terms of the
+ * Eclipse Public License v2.0 which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-v20.html
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Copyright Contributors to the Zowe Project.
+ */
+
+package org.zowe.apiml.filter;
+
+import com.netflix.eureka.registry.PeerAwareInstanceRegistryImpl;
+import lombok.RequiredArgsConstructor;
+import org.springframework.http.HttpCookie;
+import org.springframework.http.HttpHeaders;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.AuthenticationException;
+import org.springframework.security.web.server.WebFilterExchange;
+import org.springframework.security.web.server.authentication.logout.ServerLogoutHandler;
+import org.springframework.stereotype.Component;
+import org.springframework.web.server.ServerWebExchange;
+import org.zowe.apiml.handler.FailedAuthenticationWebHandler;
+import org.zowe.apiml.product.constants.CoreService;
+import org.zowe.apiml.security.common.token.TokenFormatNotValidException;
+import org.zowe.apiml.security.common.token.TokenNotValidException;
+import org.zowe.apiml.zaas.security.service.AuthenticationService;
+import reactor.core.publisher.Mono;
+
+import static org.zowe.apiml.constants.ApimlConstants.BEARER_AUTHENTICATION_PREFIX;
+import static org.zowe.apiml.security.SecurityUtils.COOKIE_AUTH_NAME;
+
+@RequiredArgsConstructor
+@Component
+public class LogoutHandler implements ServerLogoutHandler {
+
+    private final AuthenticationService authenticationService;
+    private final FailedAuthenticationWebHandler failure;
+    private final PeerAwareInstanceRegistryImpl peerAwareInstanceRegistry;
+
+    @Override
+    public Mono<Void> logout(WebFilterExchange exchange, Authentication authentication) {
+        return getTokenFromRequest(exchange.getExchange()).flatMap(token -> invalidateJwtToken(token, exchange));
+    }
+
+    public static Mono<String> getTokenFromRequest(ServerWebExchange exchange) {
+        return getCookieValue(exchange, COOKIE_AUTH_NAME)
+            .switchIfEmpty(Mono.defer(() -> getBearerTokenFromHeaderReactive(exchange)));
+    }
+
+    public static Mono<String> getCookieValue(ServerWebExchange exchange, String cookieName) {
+        return Mono.justOrEmpty(exchange)
+            .filter(ex -> ex.getRequest().getCookies().getFirst(cookieName) != null)
+            .mapNotNull(ex -> ex.getRequest().getCookies().getFirst(cookieName))
+            .map(HttpCookie::getValue);
+    }
+
+    private Mono<Void> invalidateJwtToken(String token, WebFilterExchange exchange) {
+        if (Boolean.TRUE.equals(authenticationService.isInvalidated(token))) {
```

Comment:
```suggestion
        if (Boolean.TRUE == authenticationService.isInvalidated(token)) {
```

Reply:
done

Reply:
It is not pushed?

Reply:
I'd rather have the `.equals` than check for null boolean